### PR TITLE
yaml: add support for LoadBalancer and CircuitBreaker EIPs #183

### PR DESCRIPTION
**Pull request by lburgazzoli**

### CODE REVIEW

**Comment by lburgazzoli** [comment]
@davsclaus mind having a look ?

File: camel-k-loader-yaml/camel-k-loader-yaml-common/src/main/java/org/apache/camel/k/loader/yaml/parser/LoadBalanceStepParser.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+        @JsonAlias("weighted")
+        public  void setWeighted(WeightedLoadBalancerDefinition definition) {
+            if (getLoadBalancerType() != null) {
+                throw new IllegalArgumentException("A load-balancer has already been set");
+            }
+            setLoadBalancerType(definition);;
+        }
+
+        @JsonAlias("roundRobin")
```
Comment (davsclaus) [inline camel-k-loader-yaml/camel-k-loader-yaml-common/src/main/java/org/apache/camel/k/loader/yaml/parser/LoadBalanceStepParser.java:122]: Just wonder if the alias should also have the dash variant? round-robin?

File: camel-k-loader-yaml/camel-k-loader-yaml-common/src/main/java/org/apache/camel/k/loader/yaml/parser/CircuitBreakerStepParser.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        return processor;
+    }
+
+    @YAMLNodeDefinition(reifiers = CircuitBreakerReifier.class)
+    public static final class CBDefinition {
+        public CircuitBreakerDefinition delegate = new CircuitBreakerDefinition();
+        public FBDefinition onFallback;
+
+        public List<Step> steps;
+
```
Comment (davsclaus) [inline camel-k-loader-yaml/camel-k-loader-yaml-common/src/main/java/org/apache/camel/k/loader/yaml/parser/CircuitBreakerStepParser.java:65]: When upgrading to Camel 3.3 then we also have microprofile-fault-tolerance as CB

File: camel-k-loader-yaml/camel-k-loader-yaml-common/src/main/java/org/apache/camel/k/loader/yaml/parser/LoadBalanceStepParser.java (hunk 1 above)
Comment (lburgazzoli) [reply camel-k-loader-yaml/camel-k-loader-yaml-common/src/main/java/org/apache/camel/k/loader/yaml/parser/LoadBalanceStepParser.java:122]: jackson should be able to handle both the cases but I will add a test

**Pull request merged**